File: RemoveElement/RemoveElement.cpp
#include<iostream>
#include<vector>
using namespace std;

int removeElement(vector<int>& nums, int val) {
    if (nums.size() == 0) return 0;

    int count = 0;
    for (int i = 0; i < nums.size(); i++) {
        if (nums[i] != val) {
            nums[count] = nums[i];
            count++;
        }
    }
    return count;
}


int main(){
    vector<int> nums = {3,2,2,3};
    int val = 3;
    int result = removeElement(nums, val);
    cout << result << endl;

    for (int i = 0; i < result; i++) {
        cout << nums[i] << " ";
    }
    return 0;
}

File: RemoveElement/RemoveElement.py
def removeElement(nums: list[int], val: int) -> int:
    if len(nums) == 0:
        return 0

    count = 0
    for i in range(len(nums)):
        if nums[i] != val:
            nums[count] = nums[i]
            count += 1

    return count, nums[:count]

if __name__ == "__main__":
    nums = [0,1,2,2,3,0,4,2]
    val = 2
    
    print(removeElement(nums, val)[1])